app/models/order.rb

```diff
@@ -28,32 +36,26 @@ class Order < ActiveRecord::Base
 
     return false if errors.present?
 
-    #if the inventory_tester is true for all the order items, then loop through the items one more again
-    #to adjust the amount available and the amount on hand.
-    #
-    #if there was an item that was short, dont update the order items yet.
-
     order_items.each do |item|
       item.source.on_hand -= item.quantity
       item.source.save
     end
 
     order_total = order_items.to_a.sum{|x| x.source.price * x.quantity}
-    payment = Payment.find_or_initialize_by(credit_card: user.credit_card)
+    payment = Payment.find_or_initialize_by(credit_card_id: credit_card_id, order_id: id, state: Payment::PENDING)
     payment.amount = order_total
-    payment.state = Payment::COMPLETED
-    payment.save
-
-    self.payment = payment
-    
-    if payment.persisted?
-      puts "Payment state is :  #{payment.state}"
-    else
-      return errors[:base] << "Payment failed"
+    unless payment.save
+      errors[:base] << payment.errors.map{|k, v| "Payment Error #{k}: #{v}"}
+    end
+    unless payment.make_completed
+      errors[:base] << payment.errors.map{|k, v| "Payment Error #{k}: #{v}"}
     end
+    errors.empty?
+
+  end
 
-    # TODO
-    # order.delete??
+  after_transition_to :placed do
+    UserMailer.order_email(self.id).deliver_now
```

Comment:
can you try to get ```deliver_later``` working? delivering now would cause issues if there were any problems with the third party email service. I believe you'll have to implement a queueing service such as sidekiq